### ISSUE TITLE
Issue with api/v1/applications/<appname>/manifests endpoint when using custom $values structure in ApplicationSet

### ISSUE THREAD

**Issue by RavenTouch** [argoproj/argo-cd#17607]
I am currently facing a problem with the api/v1/applications/<appname>/manifests API endpoint when using a custom file structure and $values syntax in the ApplicationSet configuration.

Issue Description:
When making a request to the /api/v1/applications/<appname>/manifests API endpoint, I encounter the following error:

```
{
  "error": "`helm template . --name-template ...... /tmp/c87c0a40-0458-4005-a521-2944975d5ed8/project/common-values/env.common.values.yaml: no such file or directory"
}
```
The error message suggests that the API is looking for a specific file (env.common.values.yaml) in a temporary directory, but this file does not exist in my Helm chart.

ApplicationSet Configuration:
I'm using an ApplicationSet that references value files using a custom $values syntax. Here's a snippet of my ApplicationSet configuration:

```
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: env-cluster-apps
spec:
  generators:
    - git:
        repoURL: <url>
        revision: master
        files:
          - path: applications/env/**/env.yaml
  template:
    metadata:
      name: env-{{path[3]}}
      namespace: argocd-prod
    spec:
      sources:
        - repoURL: <url>
          chart: project
          targetRevision: ^2.0.0
          helm:
            valueFiles:
              - $values/common-values/env.common.values.yaml
              - $values/{{path[0]}}/{{path[1]}}/cluster.common.values.yaml
              - $values/{{path[0]}}/{{path[1]}}/env.cluster.common.values.yaml
              - $values/{{path[0]}}/{{path[1]}}/{{path[2]}}/{{path[3]}}/values.yaml
              - $values/{{path[0]}}/{{path[1]}}/{{path[2]}}/{{path[3]}}/env.yaml
```
Observed Behavior:

The application deploys successfully when using this ApplicationSet configuration.
The Rollout manifest can be viewed through the ArgoCD UI and CLI without any issues.
The problem only occurs when accessing the manifest through the /api/v1/applications/<appname>/manifests API endpoint.

Expected behavior:
The API should be able to provide the manifest

Checklist:

* [x ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ x] I've included steps to reproduce the bug.
* [ x] I've pasted the output of `argocd version`.

Version: 
2.7.6
2.7.14
2.10.4

I had the same issue on these versions

**Comment by zeusal** [comment]
Hello, 

I can't see the exact problem.

What is the path of values?

<url>/common-values/env.common.values.yaml ?

About $values: 
If the path field is set in the $values source, Argo CD will attempt to generate resources from the git repository at that URL. If the path field is not set, Argo CD will use the repository solely as a source of value files.
https://argo-cd.readthedocs.io/en/stable/user-guide/multiple_sources/#helm-value-files-from-external-git-repository

**Comment by crenshaw-dev** [comment]
Small note: I think the bug would apply to a plain Application as well and is not specific to an Application produced by an ApplicationSet.

**Comment by ishitasequeira** [comment]
The issue should be fixed in 2.11 as we have added support for multiple sources to manifests api endpoint. https://github.com/argoproj/argo-cd/pull/17804
